src/pages/PaniniCreator.tsx

```diff
@@ -142,3 +147,89 @@ export default function PaniniCreator(props: PaniniCreatorProps) {
     </FormProvider>
   );
 }
+
+// possible bugs later on with generating due to HONEY_MUSTARD -> HONEY MUSTARD. Change later on if needed, also added pecorino.
+interface SandwichPayload {
+  sandwichName: string; // Max. 35 characters
+  cutlery: boolean;
+  napkins: boolean;
+  base: {
+    bread: "FULL GRAIN" | "WHEAT";
+    cheese: Array<"MOZZARELLA" | "STRACIATELLA" | "EDAM" | "GOUDA" | "PECORINO">;
+    meat: Array<"SALAMI" | "HAM" | "BACON" | "CHICKEN">;
+    dressing: Array<"OLIVE OIL" | "HONEY MUSTARD" | "RANCH" | "MAYO">;
+    vegetables: Array<
+      "SALAD" | "TOMATO" | "OBERGINE" | "BEETROOT" | "PICKLES" | "ONION" | "PEPPER" | "ASPARAGUS" | "CUCUMBER"
+    >;
+  };
+  extras: {
+    egg: Array<"FRIED EGG" | "OMELET" | "SCRAMBLED EGG">;
+    spreads: Array<"BUTTER" | "HUMMUS" | "GUACAMOLE">;
+    serving: "COLD" | "WARM" | "GRILLED";
+    topping: "SESAME" | null;
+  };
+}
+
+//for god known reasons ts won't accept this. cuz of some readonly stuff... that's why i'm using as "any".
+const literalMapper = function (val: string[]) {
+  return val.map((val) => z.literal(val)) as any;
+};
+
+const SandwichPayload = z.object({
+  sandwichName: z.string().min(1).max(35),
+  cutlery: z.boolean(),
+  napkins: z.boolean(),
+  base: z.object({
+    bread: z.union(literalMapper(breadVariants)),
```

Comment:
**nitpick:** It would be easier to do:

```ts
const breadVariants = ["WHITE", "WHEAT", "FULL GRAIN"] as const;
const breadVariantsEnum = z.enum([...breadVariants]);

...
base: z.object({
    bread: breadVariantsEnum,
...
```